tests/TestCase/ORM/QueryRegressionTest.php

```diff
@@ -155,7 +155,8 @@ public function testEagerLoadingMismatchingAliasInHasOne()
      */
     public function testEagerLoadingBelongsToManyList()
     {
-        $this->expectException(\InvalidArgumentException::class);
```

Comment:
The original exception here was the missing fields exception. That was misleading as it's not the problem the user needs to solve. This is at least an accurate exception although still not very friendly.

Figuring out how to detect and assert something "friendlier" is probably a different task.

Reply:
Can we move the expects to where they get thrown?

Reply:
Just for you.